### PR TITLE
Test circular dependancy test

### PR DESCRIPTION
**Pull request by Exclaminator**
Won't close any issues, does enforce certain properties of the code that checks for circular dependencies.

### CODE REVIEW

File: src/octopeer-github/test/main/Options/OptionsTest.ts [hunk 1]
```diff
@@ -127,7 +127,7 @@ describe("The Options class", function() {
         expect(Options.generateDefaultOptionMap()).toEqual(mockedStorageObject);
     });
 
-    it("should be the case that no circular dependacies get inserted into the code", function() {
+    describe("regarding circular depedancies, ", function() {
```
Comment (mpsijm) [inline src/octopeer-github/test/main/Options/OptionsTest.ts]: dependancies -> dependencies

File: src/octopeer-github/test/main/Options/OptionsTest.ts [hunk 2]
```diff
@@ -166,14 +166,48 @@ describe("The Options class", function() {
...
-        expect(
-            recursivelyTopologicalyRemove(
-                $.map(
-                    Options.DEPENDENCIES,
-                    (value, index) => [[index, value]]
+
+        let a = "a", b = "b", c = "c", d = "d", e = "e", f = "f";
+
+        // now for testing the functions used in the test
+        it("should be so that a empty mapping gets approved", function() {
```
Comment (mdingjan) [inline src/octopeer-github/test/main/Options/OptionsTest.ts]: a -> an

File: src/octopeer-github/test/main/Options/OptionsTest.ts [hunk 3]
```diff
@@ -127,7 +127,7 @@ describe("The Options class", function() {
         expect(Options.generateDefaultOptionMap()).toEqual(mockedStorageObject);
     });
 
-    it("should be the case that no circular dependacies get inserted into the code", function() {
+    describe("regarding circular depedancies, ", function() {
         function recursivelyTopologicalyRemove(list: [string, string][]): boolean {
```
Comment (mpsijm) [inline src/octopeer-github/test/main/Options/OptionsTest.ts]: Topologicaly -> Topologically

File: src/octopeer-github/test/main/Options/OptionsTest.ts [hunk 4]
```diff
@@ -166,14 +166,48 @@ describe("The Options class", function() {
...
+            ])).toBe(false);
+        });
+
+        it("should be so that a simple circular mapping gets rejected", function() {
+            expect(recursivelyTopologicalyRemove([
+                [a, a],
+            ])).toBe(false);
+        });
+
+        it("should be the case that there are no circular dependancies specified in the Options", function() {
```
Comment (mdingjan) [inline src/octopeer-github/test/main/Options/OptionsTest.ts]: dependancies -> dependencies

File: src/octopeer-github/test/main/Options/OptionsTest.ts (hunk 1 above)
Comment (mdingjan) [inline src/octopeer-github/test/main/Options/OptionsTest.ts]: dependancies -> dependencies

File: src/octopeer-github/test/main/Options/OptionsTest.ts [hunk 5]
```diff
@@ -166,14 +166,48 @@ describe("The Options class", function() {
...
+            expect(
+                recursivelyTopologicalyRemove(
+                    $.map(
+                        Options.DEPENDENCIES,
+                        (value, index) => [[index, value]]
+                    )
```
Comment (mpsijm) [inline src/octopeer-github/test/main/Options/OptionsTest.ts]: Can `$.map(Options.DEPENDENCIES, (value, index) => [index, value])` be one line?

**Comment by Exclaminator** [comment]
I have fixed all of your suggestions, is this now ready for merging?

**Comment by mpsijm** [comment]
Yes thanks, can be merged :D

**Comment by mdingjan** [comment]
Yes, thanks for resolving my feedback.

Will merge!

**Pull request merged**